java/app/src/main/res/layout/add_to_google_wallet_button.xml

```diff
@@ -5,12 +5,13 @@
     android:layout_width="match_parent"
     android:minWidth="267dp"
     android:layout_height="48dp"
-    android:background="@drawable/add_to_google_wallet_button_background_shape"
+    android:background="@drawable/add_to_google_wallet_button_background"
+    android:foreground="@drawable/ripple_effect"
```

Comment:
Ah, now I know what you mean with the ripple not clipping on the edges. This is easy to achieve adding the ripple to the background.
Also, the foreground property has its own intricacies that make it misbehave on v < 23 (see [official bug](https://b.corp.google.com/issues/37065042?pli=1), [forum discussions](https://stackoverflow.com/questions/52329249/why-doesnt-androidforeground-attribute-work)).

In any case, I wasn't able to see the ripple with the current implementation. Have you considered keeping it in the bg like we do for the `Add to Wallet` button?

Reply:
Strange that you couldn't see it, but I've wholly replaced the mechanism to mimic the Wallet button's and re-pushed; it should work now the same way as the wallet button.